### PR TITLE
doc: release notes: Update contents for 2.1.0

### PR DESCRIPTION
**Pull request by dleach02**
Add updated issue list and filled in some details from PRs.

Signed-off-by: David Leach <david.leach@nxp.com>

### CODE REVIEW

**Comment by dleach02** [comment]
I have made a pass at filling in details in sections without contributions from owners so please review and provide feedback.

**Review by jhedberg — APPROVED** [review]

**Review by d3zd3z — APPROVED** [review]
Confirm no known security vulnerabilities addressed in this release.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.1.rst [hunk 1]
```diff
@@ -13,10 +13,20 @@ Major enhancements with this release include:
...
+
+No security vulnerabilities received.
+
 Kernel
 ******
 
-* TBD
+* Add arch abstraction for irq_offload()
+* Add architecture interface headers and normalized APIs across all arches
+* Show faulting CPU on fatel error message
```
[the remark this answers is not in the document]
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: will do

**Review by MaureenHelm — APPROVED** [review]

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.1.rst [hunk 2]
```diff
@@ -13,10 +13,20 @@ Major enhancements with this release include:
...
+* Improve C++ compatibility
+* renamespace and automatically document arch API
```
[the remark this answers is not in the document]
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: yeah... that was cut and paste from PR. I need to look at that PR a little closer to see if there is a more meaningful phrase

**Review by dbkinder — APPROVED** [review]
Ah, that's better.  Thanks!

**Review by ioannisg — CHANGES_REQUESTED** [review]
For the Architecture release notes, I'd suppose you'd need to put ARCHes in alphabetical order.

File: doc/releases/release-notes-2.1.rst [hunk 3]
```diff
@@ -335,7 +364,34 @@ Networking
...
+  * GATT: Added a ``bt_gatt_is_subscribed()`` function to check if attribute has been subscribed
+  * GATT: Added an initializer for GATT CCC structures
+  * HCI: Added a function to get the connection handle of a connection
+  * Added ability to load CCC settings on demand to reduce memory usage
+  * Made the time to run slave connection parameters update procedure configurable
+  * Folded consecutive calls to bt_rand into one to reduce overhead
+  * Added key displacement feature for key storage
+  * Reduced severity of unavoidable warnings
+  * Added support C++20 designated initializers
+  * Mesh: Add the model extension concept as described in the Mesh Profile Specification
```
Comment (jhedberg) [inline doc/releases/release-notes-2.1.rst]: I'd add here "Mesh: Added support for acting as a Provisioner"

**Review by jukkar — APPROVED** [review]

File: doc/releases/release-notes-2.1.rst (hunk 3 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: ok

**Comment by dleach02** [comment]
> For the Architecture release notes, I'd suppose you'd need to put ARCHes in alphabetical order.

@ioannisg done

**Review by jhedberg — APPROVED** [review]

**Comment by ioannisg** [comment]
Unless I am mistaken, ARC should be before ARM.

**Comment by dleach02** [comment]
@pgielda, @nategraff-sifive, @ruuddw, @nashif, @andrewboie 

You all have been identified as needing approvals to sign off on the final release notes

**Comment by dleach02** [comment]
> Unless I am mistaken, ARC should be before ARM.

done

**Review by ruuddw — APPROVED** [review]
I'm ok for ARC. For the ARC Architecture section completness, you could add "Updates for SMP support", but it's implicitly covered by issue 19885 in the list below already.

**Review by pgielda — APPROVED** [review]

File: doc/releases/release-notes-2.1.rst [hunk 4]
```diff
@@ -2,25 +2,48 @@
...
 
 Major enhancements with this release include:
 
-* TBD
+* Normalized APIs across all architectures.
+* Expanded support for ARMv6-M architecture.
+* Added support for numerous new boards and shields
+* Added numerous new drivers and sensors.
+* Added new TCP stack implementation (experimental).
+* Numerous bug fixes to Bluetooth split software controller.
```
Comment (nashif) [inline doc/releases/release-notes-2.1.rst]: does this really belong here as a major enhancement?

File: doc/releases/release-notes-2.1.rst [hunk 5]
```diff
@@ -42,6 +65,18 @@ Architectures
...
 
+* RISC-V:
+
+  * Add support to boot multicore system
+
+* x86:
+
+  * Add basic ACPI and non-trivial memory map support
+  * Add SMP support (64-bit mode only)
+  * Add Z_ARCH_THREAD_STACK_MEMBER()
```
Comment (nashif) [inline doc/releases/release-notes-2.1.rst]: I would drop this if there are no other details, this is an internal API, actually this call does not even exist anymore.

File: doc/releases/release-notes-2.1.rst [hunk 6]
```diff
@@ -42,6 +65,18 @@ Architectures
...
+  * Inline direct ISR functions
+  * Migrate Intel64 to z_arch_switch()
```
Comment (nashif) [inline doc/releases/release-notes-2.1.rst]: lets remove that. I will give it another go after this is merged...

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.1.rst (hunk 4 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: Don't know. Was trying to capture something for this area. I'm completely open to suggestions here.

File: doc/releases/release-notes-2.1.rst (hunk 5 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: ok

File: doc/releases/release-notes-2.1.rst (hunk 6 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: ok

**Review by nategraff-sifive — APPROVED** [review]
RISC-V LGTM 👍

File: doc/conf.py [hunk 7]
```diff
@@ -528,6 +528,7 @@
     'theme_logo_only': False,
     'current_version': version,
     'versions': (("latest", "/"),
+                 ("2.1.0", "/2.1.0/"),
```
Comment (ioannisg) [inline doc/conf.py]: Since you are adding one, remove one from the bottom, so it's the same number of releases

**Review by ioannisg — APPROVED** [review]
Looks ok. I added a minor comment for the number of releases listed in conf.py, @dleach02 please, address this.

**Review by dbkinder — APPROVED** [review]

File: doc/releases/release-notes-2.1.rst (hunk 4 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: done

**Review by nashif — APPROVED** [review]

**Pull request merged**